heft-plugins/heft-jest-plugin/src/JestPlugin.ts

```diff
@@ -591,9 +607,13 @@ export default class JestPlugin implements IHeftTaskPlugin<IJestPluginOptions> {
     } else {
       // If "silent" is specified via IJestPluginOptions, that takes precedence over jest.config.json
       silent = options.silent ?? jestConfig.silent ?? false;
+      shard = options.shard;
     }
 
     const jestArgv: Config.Argv = {
+      shard,
+      coverageDirectory: options.coverageDirectory,
```

Comment:
Any recommendations on how to solve for this? Seems like there's additional complexity around the cache folder being reachable for heft cleans (AFAIK those don't clean the `.rush` folder)

Reply:
This gets extra messy because Heft's temp folder is not safe to use for the shards, since it'll get cleaned. Heft itself will need to most likely need to have the concept of shards baked in completely, so that it can scope the temp folder to be shard-specific.

Reply:
Do we want to split this work (heft/heft-jest-plugin) out into a separate PR? Sounds like there's some heft specific work that also needs to happen.

Reply:
Yeah, we should do the Heft side as a separate PR. For testing just make a project with a phase script that logs the phase input parameters.